File: app/src/main/java/com/tematikhonov/androidgeekbrainssecondapp/data/Note.java
package com.tematikhonov.androidgeekbrainssecondapp.data;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.Date;

public class Note implements Parcelable {
    private String title;
    private Date date;
    private String description;
    private boolean favorite;

    public Note(String title, Date date, String description, boolean favorite) {
        this.title = title;
        this.date = date;
        this.description = description;
        this.favorite = favorite;
    }

    protected Note(Parcel in) {
        title = in.readString();
        description = in.readString();
        date = new Date(in.readLong());
        favorite = in.readByte() != 0;
    }

    public static final Creator<Note> CREATOR = new Creator<Note>() {
        @Override
        public Note createFromParcel(Parcel in) {
            return new Note(in);
        }

        @Override
        public Note[] newArray(int size) {
            return new Note[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public Date  getDate() {
        return date;
    }

    public String getDescription() {
        return description;
    }

    public boolean isFavorite() {
        return favorite;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(description);
        dest.writeLong(date.getTime());
        dest.writeByte((byte) (favorite ? 1 : 0));
    }
}

File: app/src/main/java/com/tematikhonov/androidgeekbrainssecondapp/data/Note.kt
package com.tematikhonov.androidgeekbrainssecondapp.data

import android.os.Parcel
import android.os.Parcelable
import java.util.*

open class Note : Parcelable {
    var id: String? = null
    var title: String?
        private set
    var date: Date
        private set
    var description: String?
        private set
    var isFavorite: Boolean
        private set

    constructor(title: String?, date: Date, description: String?, favorite: Boolean) {
        this.title = title
        this.date = date
        this.description = description
        isFavorite = favorite
    }

    constructor(`in`: Parcel) {
        title = `in`.readString()
        description = `in`.readString()
        date = Date(`in`.readLong())
        isFavorite = `in`.readByte().toInt() != 0
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(description)
        dest.writeLong(date.time)
        dest.writeByte((if (isFavorite) 1 else 0).toByte())
    }

    companion object CREATOR : Parcelable.Creator<Note> {
        override fun createFromParcel(parcel: Parcel): Note {
            return Note(parcel)
        }

        override fun newArray(size: Int): Array<Note?> {
            return arrayOfNulls(size)
        }
    }

}